examples/third_party/openssl/BUILD.openssl.bazel

```diff
@@ -1,56 +1,178 @@
-"""An openssl build file based on a snippet found in the github issue:
-https://github.com/bazelbuild/rules_foreign_cc/issues/337
-"""
+# Read https://wiki.openssl.org/index.php/Compilation_and_Installation
 
 load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make")
 
-# Read https://wiki.openssl.org/index.php/Compilation_and_Installation
+# Note that this dependency name will obviously change if used outside of `rules_foreign_cc`
+load("@rules_foreign_cc_examples_third_party//openssl:openssl_repositories.bzl", "host_transitioned_foreign_cc_target")
 
 filegroup(
     name = "all_srcs",
     srcs = glob(["**"]),
 )
 
+# This wrapper is used to do some last minute configurations to the environment the ForeignCc
+# action runs under for windows. This target should not be used on any other platform.
+genrule(
+    name = "win_tool_prefix",
+    outs = ["win_tool_prefix.sh"],
+    cmd = """\
+cat << EOF > $@
+#!/bin/bash
+set -xeuo pipefail
+
+# The mingw64 target strugles with windows paths so to avoid this
+# the paths are normalized for unix if possible
+function normalize() {
+    type -t cygpath.exe > /dev/null && cygpath.exe "\\$$@" -w || echo "\\$$@"
+}
+
+# Normalize the paths to cerntain executables
+
+if [[ -n "\\$${MAKE:-}" ]]; then
+    export MAKE="\\$$(normalize "\\$${MAKE}")"
+fi
+
+if [[ -n "\\$${AR:-}" ]]; then
+    export AR="\\"\\$$(normalize "\\$${AR}")\\""
+fi
+
+if [[ -n "\\$${CC:-}" ]]; then
+    export CC="\\"\\$$(normalize "\\$${CC}")\\""
+fi
+
+if [[ -n "\\$${CXX:-}" ]]; then
+    export CXX="\\"\\$$(normalize "\\$${CXX}")\\""
+fi
+
+if [[ -n "\\$${NASM:-}" ]]; then
+    export PATH="\\$$(dirname "\\$$NASM");\\$$PATH"
+    export NASM="\\"\\$$(normalize "\\$${NASM}")\\""
+fi
+
+# Normalize the exec root for the action
+export EXT_BUILD_ROOT="\\$$(normalize "\\$${EXT_BUILD_ROOT}")"
+
+# The configure script needs to be executed with perl. If this is the first
+# argument, pass all commands to perl.
+if [[ \\$$1 == *"Configure" ]]; then
+    "\\$${EXT_BUILD_ROOT}/$(execpath @strawberry_perl//:perl)" \\$$@
+
+    # For some reason, the linker arg '/nologo' gets written out as a
+    # path to some file on disk. This corrects for that
+    sed -i "s@C:/tools/msys64/nologo@/nologo@g" \\$$BUILD_TMPDIR/makefile
+
+    # For some reason no '*.pdb' is created which (may be correct given the
+    # configuration) causes the install target to fail. To fix this the line
+    # is modified
+    # https://github.com/openssl/openssl/issues/947
+    sed -i 's/if "\\$$(SHLIBS)"==""/IF EXIST "ossl_static.pdb"/g' \\$$BUILD_TMPDIR/makefile
```

Comment:
I faced this issue also. The root cause is that the MSVC toolchain generated by Bazel adds the `/Z7` compilation flag. However, the for .pdb files to be generated, the compilation flag must be `/Zi`

See https://github.com/jheaff1/rules_foreign_cc/blob/36e298b4f325d447baaf2e692391ef322afe0007/examples/third_party/openssl/BUILD.openssl.bazel#L36

Reply:
One thing I prefer about this approach is that it doesn't override all `CFLAGS`. I think it's better to allow the rules to forward those as-is so the rules build as expected. Perhaps that approach though only leads to more confusion? Particularly if someone builds with a different configuration and it breaks 😓 but I think it should be fine.